cmd/fs-v1-metadata.go

```diff
@@ -375,6 +383,8 @@ func checkFormatSanityFSV2(fsPath string) error {
 // Check for sanity of FS format for a given version.
 func checkFormatSanityFS(fsPath string, fsFormatVersion string) (err error) {
 	switch fsFormatVersion {
+	case fsFormatV1:
```

Comment:
Explained the reason in person to @donatello 
The reason checkFormatSanityFS exists is, in the following sequence of steps:
run minio fs version 1
run minio fs version 2 (does migration)
run minio fs version 1 - this starts instead of erroring out. Hence writing in older format in .minio.sys which never gets corrected.

The sanity checking function takes care of making sure that such a corruption has not happened. If it has happened user will have to clean it up before starting minio in version 2